src/orange/g4org/SolidConverter.cc

```diff
@@ -646,9 +661,19 @@ auto SolidConverter::tet(arg_type solid_base) -> result_type
 //! Convert a torus
 auto SolidConverter::torus(arg_type solid_base) -> result_type
 {
+    CELER_LOG(warning) << "G4Torus is not fully supported; approximating with "
+                          "bounding cylinders";
     auto const& solid = dynamic_cast<G4Torus const&>(solid_base);
-    CELER_DISCARD(solid);
-    CELER_NOT_IMPLEMENTED("torus");
+    auto rmax = scale_(solid.GetRmax());
+    auto rtor = scale_(solid.GetRtor());
+
+    std::optional<Cylinder> inner;
+    inner = Cylinder{rtor - rmax, rmax};
```

Comment:
Is it possible to create a degenerate torus? If so, leave it optionaol; if not, you could write this
```suggestion
    std::optional<Cylinder> inner{std::in_place, rtor - rmax, rmax};
```
and just for reference you can also use `inner.emplace(rtor - rmax, rmax)`. Just good to know but these construct in place rather than constructing a standalone object and using the move constructor `Cylinder(Cylinder&&)`.

Reply:
Nope not optional.